lib/internal/modules/esm/resolve.js

```diff
@@ -463,6 +463,21 @@ const patternRegEx = /\*/g;
 
 function resolvePackageTargetString(
   target, subpath, match, packageJSONUrl, base, pattern, internal, conditions) {
+
+  const composeResult = (resolved) => {
+    let format;
+    try {
+      format = getPackageType(resolved);
+    } catch (err) {
+      if (err.code === 'ERR_INVALID_FILE_URL_PATH') {
+        throw new ERR_INVALID_MODULE_SPECIFIER(
+          resolved, 'must not include encoded "/" or "\\" characters', base);
```

Comment:
nit: not sure if we really care, but we could be passing the original error as cause.
```suggestion
        const er = new ERR_INVALID_MODULE_SPECIFIER(
          resolved, 'must not include encoded "/" or "\\" characters', base);
        er.cause = err;
        throw er;
```